### ISSUE TITLE
[FEATURE REQUEST] rofi will not remove options from drun

### ISSUE THREAD

**Issue by birdofscarlet** [davatorium/rofi#1989]
### Rofi version (rofi -v)

1.7.5

### Configuration

https://gist.github.com/eternaljade2001/c19c9f3604ad9f87d007f66575480899

### Theme

https://gist.github.com/eternaljade2001/5d0fe80d3ecf6427664ad4aa4fc3e037

### Timing report

n/a

### Launch command

rofi -show drun

### Step to reproduce

* Step 1: attempt to delete entry with shift-delete keys
* Step 2: entry is not deleted
* Step 3: open issue on github

### Expected behavior

the entry is removed from the menu and does not come back

### Actual behavior

the entry either will not be removed or will be removed but only until the menu is opened again

### Additional information

n/a

### Using wayland display server protocol

- [X] No, I don't use the wayland display server protocol

### I've checked if the issue exists in the latest stable release

- [X] Yes, I have checked the problem exists in the latest stable version

**Comment by DaveDavenport** [comment]
This is correct. Rofi does not support this.  (it cannot actually delete desktop files, and currently it does not have a mechanism to blacklist them).
Currently it deletes them from the frequently used history, so it won't be sorted on top anymore, but will be  the list (after re-open).

**Comment by DaveDavenport** [comment]
See https://davatorium.github.io/rofi/1.7.5/rofi.1/#drun

**Comment by birdofscarlet** [comment]
> This is correct. Rofi does not support this. (it cannot actually delete desktop files, and currently it does not have a mechanism to blacklist them). Currently it deletes them from the frequently used history, so it won't be sorted on top anymore, but will be the list (after re-open).

should i open a feature request for blacklisting files, then

**Comment by DaveDavenport** [comment]
Yes, but don't expect an implementation in short time. I'm very busy and do not have time to work on rofi.
A patch/MR that implemented the XDG standard for hiding applications from menus would be ideal. 

(edited the issue)

**Comment by birdofscarlet** [comment]
that's fine take your time, i found my own solution anyways. not sure if i should close this or not if this is a feature request post now

**Comment by Aditya-Sinha1** [comment]
> that's fine take your time, i found my own solution anyways. not sure if i should close this or not if this is a feature request post now

what's the solution ?

**Comment by DaveDavenport** [comment]
The best way is to copy the desktop file you want to hide into ~/.local/share/applications/ and set `Hidden=true`.

Or use an application editor like alacarte (if they fixed it).
